pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -45,15 +45,11 @@ func IsRealizeStateError(err error) bool {
 // backoff defines the maximum retries and the wait interval between two retries.
 func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, intentPath, entityType string) error {
 	// TODO， ask NSX if there were multiple realize states could we check only the latest one?
-	vpcInfo, err := common.ParseVPCResourcePath(intentPath)
-	if err != nil {
-		return err
-	}
 	return retry.OnError(backoff, func(err error) bool {
 		// Won't retry when realized state is `ERROR`.
 		return !IsRealizeStateError(err)
 	}, func() error {
-		results, err := service.NSXClient.RealizedEntitiesClient.List(vpcInfo.OrgID, vpcInfo.ProjectID, intentPath, nil)
+		results, err := service.NSXClient.RealizedEntitiesClient.List(intentPath, nil)
```

Comment:
Do you test this RealizedEntitiesClient is working for non-default case.
Previously, we use RealizedEntitiesClient under project/infra to check VPC related resources. 
I think we should handle both default project and non-default case using different client be checking intenPath is 
`/orgs/default/projects/default  or /orgs/default/projects/<project-id>`

Reply:
yes, it worked.